### PR TITLE
chore: upgrade to Go 1.13

### LINKED ISSUE #246: Update To Go 1.13
**Is your feature request related to a problem? Please describe.**

The eunomia operator is currently being built with Go 1.12. Go 1.12 will no longer be a supported Go version once 1.14 is released. Also, starting with operator-sdk version v.0.12.0 Go 1.13 is required.

**Describe the solution you'd like**

I would like the eunomia operator to be built using Go 1.13.

**Describe alternatives you've considered**

I haven't considered any alternatives.

**What version of eunomia are you using?**

eunomia version: v0.1.0

**Additional context**

Issue  #167 is tracking the operator-sdk version upgrade.

### PR DESCRIPTION
**Pull request by akavel**
**Description**

Upgrade eunomia codebase to Go 1.13, from no longer supported Go 1.12.

Fixes #246 

**Type of change**

* New feature (non-breaking change which adds functionality)

**Checklist**

- [x] Unit tests and e2e tests updated
- [x] Documentation updated

### CODE REVIEW

**Comment by akavel** [comment]
@seanmalloy can we globally disable the Code Coverage "target check" that "fails" for this PR? Personally I'm not convinced in value of such checks at all, but even taking that aside, the project is currently most certainly not in a state where it'd be ready for such targets yet.

File: .travis.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ sudo: required
 language: go
 go_import_path: github.com/KohlsTechnology/eunomia
 go:
-- "1.12.x"
+- "1.13.x"
 env:
   global:
   - GO111MODULE=on
```
Comment (sbar95) [inline .travis.yml]: Do we need it with go 1.13?

Comment (akavel) [reply .travis.yml]: That's an awesome question, thanks for raising it! I'll be looking into this, you may be right. ❤️

**Comment by akavel** [comment]
As to DEVELOPMENT.md, I believe [one of the commits in this PR already updates it](https://github.com/KohlsTechnology/eunomia/pull/314/commits/fdd0a5c75762d0e1d78a3ff499bdd1ae59439393#diff-f785e9e46b9afd2567df7a658338db17); if there are some specific places I missed, could you help me see them?

**Comment by sbar95** [comment]
@akavel My bad, it's been updated, just somehow I didn't see it :/

**Comment by seanmalloy** [comment]
> can we globally disable the Code Coverage "target check" that seems to "fail" for this PR.

Yes, I can turn it off. I'm not sure how it got turned on. I did not turn it on. The codecov.io stuff if currently intended as a gentle reminder to write unit tests. :-)

**Comment by seanmalloy** [comment]
@akavel I investigated if there is any easy way to adjust the codecov.io settings, but I was not able to figure it out in 5 minutes. My guess this is a new feature created by codecov.io. Only the Travis CI check is required.

I created #316 for tracking adjusting the codecov.io settings.

**Pull request merged**